user_tools/src/spark_rapids_tools/configuration/tools_config.py

```diff
@@ -34,9 +35,15 @@ class ToolsConfig(BaseModel):
         examples=['1.0'],
         le=1.0,  # minimum version compatible with the current tools implementation
         ge=1.0)
-    runtime: ToolsRuntimeConfig = Field(
+
+    runtime: Optional[ToolsRuntimeConfig] = Field(
+        default=None,
         description='Configuration related to the runtime environment of the tools.')
 
+    distributed_tools: Optional[DistributedToolsConfig] = Field(
```

Comment:
Having `runtime` on the same level as `distributed_tools` makes it confusing.
One would expect that `runtime` is generic property that applies for all submission modes.
I suggest that we have a `DistributedToolsRuntimeConfig` that extends `ToolsRuntimeConfig` or we have an abstract `RuntimeConfig` that gets extended by `local` and `Distributed` implementation.
In that case the format of the file will be more consistent for both modes.

Reply:
From offline discussion, updated the config file format to be as follows:

```yaml
api_version: '1.1'
runtime:
  dependencies:
    - name: my-spark350
      uri: https:///archive.apache.org/dist/spark/spark-3.5.0/spark-3.5.0-bin-hadoop3.tgz
      dependency_type:
        dep_type: archive
        # for tgz files, it is required to give the subfolder where the jars are located
        relative_path: jars/*
submission:
  remote_cache_dir: 'hdfs:///tmp/spark_rapids_distributed_tools_cache'
  spark_properties:
    - name: 'spark.executor.memory'
      value: '20g'
```